test/directory.dart

```diff
@@ -0,0 +1,51 @@
+import 'dart:io';
+
+import 'package:test/test.dart';
+import 'package:dartx/dartx_io.dart';
+
+void main() {
+  group('Directory', () {
+    test('contains', () async {
+      final thisDir = Directory.current;
+      final parentDir = thisDir.parent;
+      final parentParentDir = parentDir.parent;
+      File('foo.txt').createSync();
+      final file = File('foo.txt');
+
+      expect(await parentDir.contains(thisDir), true);
+      expect(await parentDir.contains(thisDir, recursive: true), true);
+      expect(await thisDir.contains(parentDir), false);
+      expect(await thisDir.contains(thisDir), false);
+      expect(await parentParentDir.contains(thisDir), false);
+      expect(await parentParentDir.contains(thisDir, recursive: true), true);
+
+      expect(await thisDir.contains(file), true);
+      expect(await thisDir.contains(file, recursive: true), true);
+      expect(await parentDir.contains(file), false);
+      expect(await parentDir.contains(file, recursive: true), true);
+
+      file.deleteSync();
```

Comment:
wrap the deletion in `addTearDown` and place it right after creating the file.

Reply:
the temporary directory is now deleted recursively inside `addTearDown`